src/goto-symex/field_sensitivity.cpp

```diff
@@ -0,0 +1,253 @@
+/*******************************************************************\
+
+Module: Field-sensitive SSA
+
+Author: Michael Tautschnig
+
+\*******************************************************************/
+
+#include "field_sensitivity.h"
+
+#include <util/arith_tools.h>
+#include <util/c_types.h>
+#include <util/simplify_expr.h>
+#include <util/std_expr.h>
+
+#include "goto_symex_state.h"
+#include "symex_target.h"
+
+field_sensitivityt::field_sensitivityt(
+  const namespacet &ns,
+  symex_targett &target)
+  : ns(ns), target(target)
+{
+}
+
+void field_sensitivityt::apply(exprt &expr, bool write) const
+{
+  if(!run_apply)
+    return;
```

Comment:
I think this may no longer be necessary, as `field_assignments_rec` does not use `apply`

Reply:
`field_assignments_rec` calls `rename`, which turn will call `apply`. And I don't think it should be upon the caller to know whether to call `apply`, because that would mean the implementation details of field sensitivity are spilled all over the place. So I'd really keep this as is.